framework/src/modules/chain/components/storage/entities/round_delegates.js

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright © 2018 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+
+'use strict';
+
+const path = require('path');
+const assert = require('assert');
+const {
+	entities: { BaseEntity },
+	utils: {
+		filterTypes: { NUMBER },
+	},
+} = require('../../../../../components/storage');
+
+const sqlFiles = {
+	create: 'round_delegates/create.sql',
+	delete: 'round_delegates/delete.sql',
+	getRoundDelegates: 'round_delegates/get_round_delegates.sql',
+};
+
+/**
+ * Round
+ * @typedef {Object} Round
+ * @property {string} address
+ * @property {number} amount
+ * @property {string} delegate
+ * @property {number} round
+ */
+
+/**
+ * Round Filters
+ * @typedef {Object} filters.Round
+ * @property {number} [round]
+ * @property {number} [round_eql]
+ * @property {number} [round_ne]
+ * @property {number} [round_gt]
+ * @property {number} [round_gte]
+ * @property {number} [round_lt]
+ * @property {number} [round_lte]
+ * @property {number} [round_in]
+ * @property {json} [delegatePublicKeys]
+ */
+
+class RoundDelegates extends BaseEntity {
+	/**
+	 * Constructor
+	 * @param {BaseAdapter} adapter - Adapter to retrieve the data from
+	 * @param {filters.Round} defaultFilters - Set of default filters applied on every query
+	 */
+	constructor(adapter, defaultFilters = {}) {
+		super(adapter, defaultFilters);
+
+		this.addField('delegatePublicKeys', 'string');
+		this.addField('round', 'number', { filter: NUMBER });
+
+		this.extendDefaultOptions({ sort: '' });
+
+		this.sqlDirectory = path.join(path.dirname(__filename), '../sql');
+
+		this.SQLs = this.loadSQLFiles('dpos', sqlFiles, this.sqlDirectory);
+	}
+
+	async get(round) {
+		const result = await this.adapter.executeFile(this.SQLs.getRoundDelegates, {
+			round,
+		});
+		return result[0].delegatePublicKeys;
+	}
+
+	/**
+	 * Create round object
+	 *
+	 * @param {Object} data
+	 * @param {Object} [_options]
+	 * @param {Object} [tx] - Transaction object
+	 * @return {null}
+	 */
+	// eslint-disable-next-line no-unused-vars
+	create({ round, delegatePublicKeys }, _options = {}, tx = null) {
+		assert(round, 'Round must be an object');
+		assert(
+			Array.isArray(delegatePublicKeys),
+			'delegatePublicKeys must be an array of strings'
+		);
+
+		const attributes = Object.keys(this.fields).filter(
+			fieldname => fieldname !== 'id'
```

Comment:
We don't have an id field in the database table corresponding to this entity. So this check is unnecessary